corehq/apps/domain/tests/test_auth_decorators.py

```diff
@@ -0,0 +1,38 @@
+from django.test import SimpleTestCase
+
+
+CHECK_FAILED = 'check failed'
+
+
+def challenge(check):
+    def decorator(view):
+        def wrapper(request, *args, **kwargs):
+            auth = check(request)
+            if auth:
+                return view(request, *args, **kwargs)
```

Comment:
Yeah, it's a good question. At one point it was doing that and then I moved the approach of monkey patching the request on the user (I can't remember why). It would be clearer to do it in the decorator, so I'll take a peek and if it's not prohibitively annoying I'll change it.

Reply:
this did end up being a pain, but was far enough along by the time I realized all the issues that I went for it anyways. 28678389e84f30dabb82621c7a332030c015af8b